core/kernel/tee_ta_manager.c

```diff
@@ -416,18 +438,22 @@ TEE_Result tee_ta_close_session(struct tee_ta_session *csess,
 	ctx = sess->ctx;
 	DMSG("Destroy session");
 
+	if (!ctx) {
+		/* The context is already destroyed */
+		tee_ta_destroy_session(sess, open_sessions);
+		return TEE_SUCCESS;
```

Comment:
Should we set `*err = TEE_ORIGIN_TEE;` here?

Reply:
`tee_ta_close_session` doesn't return origin code. It is done by the callee `entry_close_session`.